src/bin/pgcopydb/pgsql_timeline.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * src/bin/pgcopydb/pgsql_timeline.h
+ *	 API for sending SQL commands about timelines to a PostgreSQL server
+ */
+#ifndef PGSQL_TIMELINE_H
+#define PGSQL_TIMELINE_H
+
+#include "pgsql.h"
+#include "schema.h"
+
+typedef struct ParseTimelineHistoryContext
+{
+	uint32_t currentTimeline;
+	DatabaseCatalog *catalog;
+} ParseTimelineHistoryContext;
+
+
+typedef struct TimelineHistoryIterator
+{
+	const char *filename;
+	FileLinesIterator *fileIterator;
+	TimelineHistoryEntry *entry;
+	uint64_t prevend;
+	uint32_t currentTimeline;
+} TimelineHistoryIterator;
+
+/* iterate over a file one line at a time */
+typedef bool (TimelineHistoryFun)(void *context, TimelineHistoryEntry *entry);
+
+/* pgsql_timeline.c */
+bool pgsql_identify_system(PGSQL *pgsql, IdentifySystem *system,
+						   char *cdcPathDir);
+bool timeline_iter_history(char *filename,
+						   ParseTimelineHistoryContext *context,
+						   TimelineHistoryFun *callback);
+
+bool timeline_iter_history_init(TimelineHistoryIterator *iter);
+bool timeline_iter_history_next(TimelineHistoryIterator *iter);
+bool timeline_iter_history_finish(TimelineHistoryIterator *iter);
+
+bool timeline_history_add_hook(void *context, TimelineHistoryEntry *entry);
```

Comment:
I'm not fond of the name. Not a showstopper. Will think. The idea is to describe what the hook does, not where we call it...

Reply:
I ended up removing this completely. Relevant discussion at other comment: https://github.com/dimitri/pgcopydb/pull/848#discussion_r1698482611